src/helpers/workspace.ts

```diff
@@ -17,6 +25,49 @@ export function getWorkspaceRoot(ignoreException: boolean = false): string | und
   return workspaceRoot;
 }
 
+export async function getWorkspaces(): Promise<TruffleWorkspace[]> {
+  const workspaces: TruffleWorkspace[] = [];
+
+  await Promise.all(
+    workspace.workspaceFolders!.map(async (ws) => {
+      workspaces.push(...(await getWorkspaceFiles(ws.uri.fsPath)));
+    })
+  );
+
+  if (workspaces === undefined) {
```

Comment:
how can `workspaces` be ever undefined when you make it an array initialiser up top? Should you raise an error when its empty/0?